### PR TITLE
Rename to `n_`

### PR DESCRIPTION
**Pull request by jhprinz**
This renames all variable representing the number of something to `n_`.

Including

- [x] `n_atoms`
- [x] `n_spatial`
- [x] `n_steps_per_frame`
- [x] `n_frames`

This also includes some cleanup in `dynamics_engine.py` and `trajectory.py`

### CODE REVIEW

**Comment by jhprinz** [comment]
Ready for review and merge...

The only effective change is. That every engine will now support getting the snapshot dimensions by `n_[dimensionname]` automatically. See `dynamics_engine.py` for the little trick in `__getattr__`

File: examples/ipython/langevin_integrator_check.ipynb [hunk 1]
```diff
@@ -122,7 +122,7 @@
    "outputs": [],
    "source": [
     "for i in range(nframes):\n",
-    "    # generate the next frame (which is sim.nsteps_per_iteration timesteps)\n",
+    "    # generate the next frame (which is sim.n_steps_per_iteration timesteps)\n",
```
Comment (dwhswenson) [inline examples/ipython/langevin_integrator_check.ipynb]: `n_steps_per_iteration` => `n_steps_per_frame` (in a comment so it doesn't really matter, but let's go ahead and fix that now)

Comment (jhprinz) [inline examples/ipython/langevin_integrator_check.ipynb]: Sure. strange the replace somehow missed that...

Comment (jhprinz) [inline examples/ipython/langevin_integrator_check.ipynb]: Weird. I had changed that actually (this is why I could not find `nsteps` anymore, but didnot commit it...

Comment (dwhswenson) [inline examples/ipython/langevin_integrator_check.ipynb]: problem isn't `nsteps`=>`n_steps`, it is `per_iteration`=>`per_frame`. Old (or possibly never actually used?) name for the same concept -- just used a comment, but we should make the comments match code when we catch a discrepancy!

**Comment by dwhswenson** [comment]
One line note about fixing the problem that an old comment refers to `n_steps_per_iteration`, which is should refer to `n_steps_per_frame`. Otherwise, it all looks good to me. Will merge after that fix and after it passes tests.

File: examples/ipython/langevin_integrator_check.ipynb (hunk 1 above)
Comment (jhprinz) [inline examples/ipython/langevin_integrator_check.ipynb]: Sorry, didnot see that... Just a second

**Comment by jhprinz** [comment]
Done...

**Pull request merged**